Requirements.md

```diff
@@ -1,64 +1,287 @@
+(rewrite all FP expressions over BV terms).
 
-### Theories support
-Bv, Arrays, Ints, FP, Q-
+### CVC5 solver support
 
-### Unsupported features
-If some solver doesn't support some theory (like Bitvec) or some feature(unsat-core) we need to throw 
-specialized exception. Exception should contain recommendation to switch to other solver.
+[CVC5](https://github.com/cvc5/cvc5) is
+a [well performing](https://smt-comp.github.io/2022/results/qf-bitvec-single-query) SMT solver.
 
-### Expression naming
-Add debug information into expression for future visualization
+CVC5 has native support for all theories,
+listed in [KSMT supported theories](#basic-theories-support)
+and provides all the functionality, listed in [SMT solver features](#smt-solver-support).
 
-### External process
+### External process runner
+
+Run solvers in separate processes to preserve user applications stability.
+
+SMT solvers are implemented via native libraries and usually have the following issues:
+
+1. Timeout ignore. SMT solver may hang in a long-running operation before reaching a checkpoint.
+   Since solver is a native library, there is no way to interrupt it.
+2. Solvers are usually research projects with a bunch of bugs, e.g. pointer issues. Such
+   errors lead to the interruption of the entire process, including the user's app.
+
+Currently, we have a [process runner implementation](ksmt-runner), that allows us to
+force the solver to respect timeouts and survive after native errors.
 
 ### Portfolio solver
 
+Various SMT solvers usually show different performance on a same SMT formula.
+[Portfolio solving](https://arxiv.org/abs/1505.03340) is a simple idea of
+running different solvers on the same formula simultaneously and waiting only
+for the first result.
+
+This approach can be implemented on top of [external process runner](#external-process-runner).
+
+### Solver configuration API
+
+Extend SMT solver interface with an option to
+pass solver specific parameters to the underlying solver.
+
+For implementation details, check out [corresponding PR](https://github.com/UnitTestBot/ksmt/pull/34).
+
+
+### Deployment
+
+Deliver KSMT to end users. 
+
+Currently, we use [JitPack](https://jitpack.io) for deployment.
+
+Use Maven Central in the future.
+
+### Expression simplification / evaluation
+
+Implement simplification rules for KSMT expressions, and apply it to:
+1. Expression simplification
+2. Lightweight eager simplification during expressions creation
+3. Expression evaluation wrt model (special case of simplification)
+
+
 ### Performance tests
-Describe how to write performance tests that measure overhead of KSMT for typical operations.
-After done, add results here
 
-### Solver Native API support
-Don't use Z3Java API
+Measure overhead of the following operations for all supported solvers:
+1. Expression interanlization (conversion from KSMT to solver native). The most used and the most expensive operation.
```

Comment:
Errata in word "internalization". BTW, As there are no common term for it, I think "translation" term is better suites for conversion from KExpr to z3expr then "internalization"